adobe/spectrum-css
# build: upgrade to yarn 4.1.1

## Description

Why upgrade from yarn 1.x (aka classic)?

> **Stability**: Yarn Modern came after years of experience with maintaining Classic; its design was informed by the shortcomings we saw, and as a result the software is much more stable than it ever was.
> 
> **New features**: Yarn Modern provides many new features that didn't exist in Yarn 1.x - or any other package managers for that matter. As an example, [Constraints](https://yarnpkg.com/features/constraints) are exclusive to Yarn Modern.
> 
> **Flexibility**: Yarn Modern supports all three installation strategies: Yarn PnP, node_modules, and via a pnpm-like content-addressed cache. No matter which one you prefer, you have it at your disposal.
> 
> **Extensibility**: Yarn Modern's architecture allows you to build your own features as you need it. No need to wait for us to implement this feature you hope for - you can now implement it yourself, according to your own specs, and use it straight away! Focused workspaces, custom installs, project validation, ...
> 
> **Future proof**: Yarn Modern was built after we started to see how difficult it was to build new features on Yarn Classic, with most changes having unpredictable consequences. This plateau has been solved, as evidenced by the slate of features that we released in the past few major releases.

My reasons for support an upgrade to Yarn Modern is it's tooling around monorepo management (see [constraints](https://yarnpkg.com/features/constraints), [versioning](https://yarnpkg.com/features/release-workflow), [interactive upgrades](https://yarnpkg.com/cli/upgrade-interactive), and a strong set of [command-line tools](https://yarnpkg.com/cli)).

## How and where has this been tested?

Please tag yourself on the tests you've marked complete to confirm the tests have been run by someone other than the author.

### Validation steps

Setting up your environment:
1. `corepack enable` [more detailed instructions](https://yarnpkg.com/getting-started/install)
2. `nvm use` to ensure you are using the right version of node
3. `yarn install` - expect the output formatting to be quite colorful compared with yarn 1

### Regression testing

Validate:

1. No command line regressions: @mdt2 
  - [x] `yarn install`
  - [x] `yarn build`
  - [x] `yarn dev`
  - [x] `yarn start`
2. The documentation pages for at least two other components are still loading, including:
3. If components have been modified, VRTs have been run on this branch:

## To-do list
